### ISSUE TITLE
RF Classifier Feature Selectors can raise ValueError: Unknown label type: 'unknown' if nullable types are used 

### ISSUE THREAD

**Issue by tamargrey** [alteryx/evalml#3975]
```python
    import woodwork as ww
    X = pd.DataFrame({
            "nullable bool col": [True, False, False, True, True] * 4,
            "nullable int col": [0, 1, 2, 0, 3] * 4,
    })
    X.ww.init(logical_types={"nullable bool col": "BooleanNullable", "nullable int col": "IntegerNullable"})
    y = pd.Series([True, False]*(len(X)//2), dtype="boolean")
    # For regression problems, need to have an IntegerNullable column to see the error
    y = ww.init_series(y, logical_type="BooleanNullable")

    # Either the Selector or SelectFromModel produce this error 
    component = RFClassifierRFESelector()
    # component = RFClassifierSelectFromModel()

 
    component.fit(X, y)
```
This will not currently be seen in automl search because of the replace nullable types component, but we should consider adding nullable handling into the component class itself so that it can independently support nullable types.

Note this is likely related to https://github.com/alteryx/evalml/issues/3923, https://github.com/alteryx/evalml/issues/3922 , and https://github.com/alteryx/evalml/issues/3910 , which all stem from the inability of sklearn's type_of_target to assign a proper type to nullable data

**Comment by tamargrey** [comment]
Appears to be fixed by updating to sklearn 1.2.1

**Comment by tamargrey** [comment]
Closing with merge of https://github.com/alteryx/evalml/pull/3983
